robosuite/models/tasks/table_top_task.py

```diff
@@ -1,36 +1,66 @@
-from robosuite.models.tasks import Task, UniformRandomSampler
+import collections
+from copy import deepcopy
+
+from robosuite.models.world import MujocoWorldBase
+from robosuite.models.tasks import UniformRandomSampler
+from robosuite.models.objects import MujocoGeneratedObject, MujocoXMLObject
 from robosuite.utils.mjcf_utils import new_joint, array_to_string
 
 
-class TableTopTask(Task):
+class TableTopTask(MujocoWorldBase):
     """
-    Creates MJCF model of a tabletop task.
+    Creates MJCF model for a task performed on a table top (or similar surface).
 
-    A tabletop task consists of one robot interacting with a variable number of
-    objects placed on the tabletop. This class combines the robot, the table
-    arena, and the objetcts into a single MJCF model.
+    A manipulation task consists of one or more robots interacting with a variable number of
+    objects placed on a table. This class combines the robot(s), the arena, and the objects 
+    into a single MJCF model.
     """
 
-    def __init__(self, mujoco_arena, mujoco_robots, mujoco_objects, initializer=None):
+    def __init__(
+        self, 
+        mujoco_arena, 
+        mujoco_robots, 
+        mujoco_objects, 
+        visual_objects=None, 
+        initializer=None,
+    ):
         """
         Args:
             mujoco_arena: MJCF model of robot workspace
             mujoco_robots: MJCF model of robot model(s) (list)
             mujoco_objects: a list of MJCF models of physical objects
```

Comment:
would it make more sense to rename this to collision_objects to differentiate between this and the visual objects?

Reply:
we could certainly do that, but I left it as is for backwards compatibility because we have this `mujoco_objects` dictionary as a property in all environments too. We can certainly change it if both you and @yukezhu think it should be done - let me know